cpp/velox/utils/VeloxBatchResizer.cc

```diff
@@ -0,0 +1,111 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "VeloxBatchResizer.h"
+
+namespace gluten {
+
+namespace {
+class SliceRowVector : public ColumnarBatchIterator {
+ public:
+  SliceRowVector(int32_t maxOutputBatchSize, facebook::velox::RowVectorPtr in)
+      : maxOutputBatchSize_(maxOutputBatchSize), in_(in) {
+    GLUTEN_CHECK(in->size() > maxOutputBatchSize, "Invalid state");
+  }
+
+  std::shared_ptr<ColumnarBatch> next() override {
+    int32_t remainingLength = in_->size() - cursor_;
+    GLUTEN_CHECK(remainingLength >= 0, "Invalid state");
+    if (remainingLength == 0) {
+      return nullptr;
+    }
+    int32_t sliceLength = std::min(maxOutputBatchSize_, remainingLength);
```

Comment:
Why use `maxOutputBatchSize_` but not default batch size (4096)?

Reply:
This could be a good point. I was worrying if batch slicing brings extra overhead when size is just slightly larger than 4096, say 5000.

I could raise a PR later to change the value to a more reasonable one. Before that I think it's just OK to use a large value since it aligns with current code (which doesn't split at all) as well.